bridge/client/app/_components/_dialogs/ktb-deletion-dialog/ktb-deletion-dialog.component.spec.ts

```diff
@@ -0,0 +1,163 @@
+
+
+describe('KtbDeletionDialogComponent', () => {
+  let component: KtbDeletionDialogComponent;
+  let fixture: ComponentFixture<KtbDeletionDialogComponent>;
+  const dialogData = {name: 'sockshop', type: DeleteType.PROJECT};
+  let eventService: EventService;
+  const dialogRefMock = {
+    close: () => { }
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [KtbDeletionDialogComponent],
+      imports: [AppModule, MatDialogModule],
+      providers: [
+        EventService,
+        {provide: MAT_DIALOG_DATA, useValue: dialogData},
+        {provide: MatDialogRef, useValue: dialogRefMock}
+      ]
+    })
+      .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(KtbDeletionDialogComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should be a valid form when input matches the given name', () => {
+    // given
+    const input = component.deletionConfirmationControl;
+
+    // when
+    input.setValue('sockshop');
+    component.deletionConfirmationForm.updateValueAndValidity();
+
+    // then
+    expect(component.deletionConfirmationForm.valid).toBeTrue();
+  });
+
+  it('should be an invalid form when input is empty', () => {
+    // given
+    const input = component.deletionConfirmationControl;
+
+    // when
+    input.setValue('');
+    component.deletionConfirmationForm.updateValueAndValidity();
+
+    // then
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+  });
+
+  it('should be an invalid form when name and input do not match', () => {
+    // given
+    const input = component.deletionConfirmationControl;
+
+    input.setValue('sock');
+    component.deletionConfirmationForm.updateValueAndValidity();
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+
+    input.setValue('$ock');
+    component.deletionConfirmationForm.updateValueAndValidity();
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+
+    input.setValue('1');
+    component.deletionConfirmationForm.updateValueAndValidity();
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+
+    input.setValue('Sockshop');
+    component.deletionConfirmationForm.updateValueAndValidity();
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+
+    input.setValue('sockshoP');
+    component.deletionConfirmationForm.updateValueAndValidity();
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+
+    input.setValue('sOckshop');
+    component.deletionConfirmationForm.updateValueAndValidity();
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+
+    input.setValue('');
+    component.deletionConfirmationForm.updateValueAndValidity();
+    expect(component.deletionConfirmationForm.invalid).toBeTrue();
+  });
```

Comment:
Because we have a lot of duplicated code here I would suggest to create an array with the values and iterate over them. Then it will also be easier to add or remove new values

Reply:
Done :)